src/reduxTypes.js

```diff
@@ -128,27 +130,19 @@ export type FetchingState = $ReadOnly<{|
  * `state.messages`.
  */
 export type FlagsState = $ReadOnly<{|
-  // Flags that are currently documented in the API:
-  //   https://zulip.com/api/update-message-flags#available-flags
-  // i.e., those found in `UserMessageFlag`.
   read: {| +[messageId: number]: true |},
   starred: {| +[messageId: number]: true |},
   collapsed: {| +[messageId: number]: true |},
   mentioned: {| +[messageId: number]: true |},
   wildcard_mentioned: {| +[messageId: number]: true |},
   has_alert_word: {| +[messageId: number]: true |},
   historical: {| +[messageId: number]: true |},
-
-  // Flags not documented in the API.
-  // TODO(server): Do these ever exist?  Did they use to and get removed?
-  summarize_in_home: {| +[messageId: number]: true |},
-  summarize_in_stream: {| +[messageId: number]: true |},
-  force_expand: {| +[messageId: number]: true |},
-  force_collapse: {| +[messageId: number]: true |},
-  is_me_message: {| +[messageId: number]: true |},
 |}>;
 
-export type FlagName = $Keys<FlagsState>;
+// The flags in FlagsState correspond with those in the server API,
+// as expressed by UserMessageFlag.
+(f: UserMessageFlag): $Keys<FlagsState> => f;
+(f: $Keys<FlagsState>): UserMessageFlag => f;
```

Comment:
Or, if you think it's a bit better (more explicit about what we're doing):

```js
typesEquivalent<$Keys<FlagsState>, UserMessageFlag>();
```

I think that gives the same result; we have a comment near `typesEquivalent`'s definition:
```js
// Another formulation that's believed to give the same results:
//   (x: T): U => x;
//   (x: U): T => x;
// Those together should be accepted exactly when typesEquivalent<T, U>() is.
```

Reply:
Good idea, thanks.